File: CodeWars/.py/8kyu-Is-the-string-uppercase.py
# Is the string uppercase?
# Task
# Create a method to see whether the string is ALL CAPS.

# Examples (input -> output)
# "c" -> False
# "C" -> True
# "hello I AM DONALD" -> False
# "HELLO I AM DONALD" -> True
# "ACSKLDFJSgSKLDFJSKLDFJ" -> False
# "ACSKLDFJSGSKLDFJSKLDFJ" -> True
# In this Kata, a string is said to be in ALL CAPS whenever it does not contain any lowercase letter so any string containing no letters at all is trivially considered to be in ALL CAPS.

def gen_test_case(inp, res):
    test.assert_equals(is_uppercase(inp), res, inp)

test.describe("Basic Tests")

gen_test_case("c", False)
gen_test_case("C", True)
gen_test_case("hello I AM DONALD", False)
gen_test_case("HELLO I AM DONALD", True)
gen_test_case("$%&", True)

# my code

# 
# @param: string
# @return: boolean - if argument string is all uppercase or not
# 
def is_uppercase(inp):
    return inp==inp.upper()

File: CodeWars/.cs/8kyu-IsTheStringUppercase.cs
// Is the string uppercase?
// Task
// Create a method to see whether the string is ALL CAPS.

// Examples (input -> output)
// "c" -> False
// "C" -> True
// "hello I AM DONALD" -> False
// "HELLO I AM DONALD" -> True
// "ACSKLDFJSgSKLDFJSKLDFJ" -> False
// "ACSKLDFJSGSKLDFJSKLDFJ" -> True
// In this Kata, a string is said to be in ALL CAPS whenever it does not contain any lowercase letter so any string containing no letters at all is trivially considered to be in ALL CAPS.

using NUnit.Framework;
using System;

public class Tests
{
  [Test]
  [TestCase("c", ExpectedResult=false)]
  [TestCase("C", ExpectedResult=true)]
  [TestCase("hello I AM DONALD", ExpectedResult=false)]
  [TestCase("HELLO I AM DONALD", ExpectedResult=true)]
  [TestCase("ACSKLDFJSgSKLDFJSKLDFJ", ExpectedResult=false)]
  [TestCase("ACSKLDFJSGSKLDFJSKLDFJ", ExpectedResult=true)]
  public static bool FixedTest(string text)
  {
    return text.IsUpperCase();
  }
}

// my code

public static class StringExtensions
{
  public static bool IsUpperCase(this string text) => text == text.ToUpper();
}